h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/BoosterParms.java

```diff
@@ -41,7 +42,7 @@ public static BoosterParms fromMap(Map<String, Object> map) {
    */
   private static Map<String, Object> localizeDecimalParams(final Map<String, Object> params) {
     Map<String, Object> localized = new HashMap<>(params.size());
-    final NumberFormat localizedNumberFormatter = DecimalFormat.getNumberInstance();
+    final NumberFormat localizedNumberFormatter = DecimalFormat.getNumberInstance(Locale.ENGLISH);
```

Comment:
the issue here is that all tests will pass regardless if this is correct or wrong because all the test servers are already running US_EN locale

Reply:
The test explicitely changes locale just because of that. For a tiny while.

Reply:
Can we verify it also affect the native code? And not just java?

Reply:
I verified it manually by setting locale to english, undo the changes in this PR and ran this test. It failed as well.

Reply:
The setLocale call only affects the JVM. I thought you are asking about different combinations of Locale and formatter set to English locale explicitly.

Reply:
yes, I am not sure this test proves that xgb will work fine when the JVM is started on a different locale

Reply:
It does. The default, non-arg method looks like this:

```
    public final static NumberFormat getNumberInstance() {
        return getInstance(Locale.getDefault(Locale.Category.FORMAT), NUMBERSTYLE);
    }
```

And we're setting the locale temporarily by using the following method in Locale class:

```
    public static synchronized void setDefault(Locale newLocale) {
        setDefault(Category.DISPLAY, newLocale);
        setDefault(Category.FORMAT, newLocale);
        defaultLocale = newLocale;
    }
```

We should be fine. Setting just FORMAT could be theoretically sufficient, but I see no point in having two different settings on one JVM at one time. And of course, if there are errors and those are printed, the output could be misleading.

Reply:
It doesn't test the whole machine with native XGBoost running on different locale. Of course.